sdk/go/common/resource/plugin/context.go

```diff
@@ -113,14 +115,20 @@ func (ctx *Context) Request() context.Context {
 	}
 	c = opentracing.ContextWithSpan(c, ctx.tracingSpan)
 	c, cancel := context.WithCancel(c)
+	ctx.lock.Lock()
 	ctx.cancelFuncs = append(ctx.cancelFuncs, cancel)
+	ctx.lock.Unlock()
 	return c
 }
 
 // Close reclaims all resources associated with this context.
 func (ctx *Context) Close() error {
 	defer func() {
-		for _, cancel := range ctx.cancelFuncs {
+		ctx.lock.Lock()
+		cancelFuncs := ctx.cancelFuncs
+		ctx.cancelFuncs = nil
+		ctx.lock.Unlock()
```

Comment:
This bit is tricky enough that we should have a comment.
Perhaps something like:

```
// It's possible that `Request()` will be called
// and try to append to the slice
// while `Close()` is running.
// Leave a fresh empty slice behind
// so that the list of cancelFuncs doesn't change
// while we're doing that.
```